### PR TITLE
Issue/1982 add command to pause agent

### LINKED ISSUE #1982: Add cli command to pauze an agent
@wouterdb commented on [Fri Mar 20 2020](https://github.com/inmanta/inmanta-telco/issues/39)

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Added a command to the inmanta-cli to pause and unpause an agent.

closes #1982 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Comment by arnaudsjs** [comment]
@wouterdb Shouldn't we also have a command to pause/unpause all agents in a specific environment?

**Comment by wouterdb** [comment]
That would make sense! 

(the big red button)

**Comment by arnaudsjs** [comment]
Which command would be best to pause all agents in a certain environment:

1) `inmanta-cli agent pause-all -e <env>`     --\> A different command
2) `inmanta-cli agent pause -e <env>`          --\> Omitting the --agent option
3) `inmanta-cli agent pause -e <env> --all`   --\> Add the --all flag to prevent that someone accidentally pauses all agents by forgetting the --agent flag
4) ...

File: src/inmanta/main.py [hunk 1]
```diff
@@ -446,9 +446,47 @@ def agent_list(client: Client, environment: str) -> None:
...
 
-    print_table(["Agent", "Environment", "Last fail over"], data)
+    print_table(["Agent", "Environment", "Paused", "Last fail over"], data)
+
+
+@agent.command(name="pause",)
+@click.option("--environment", "-e", help="The environment to use", required=True)
+@click.option(
+    "--agent",
+    help="The name of the agent to pause. When this option is omitted, all agents in the given environment will be " "paused.",
```
Comment (wouterdb) [inline src/inmanta/main.py:458]: I think the option with explicit --all would be safer.

**Pull request merged**